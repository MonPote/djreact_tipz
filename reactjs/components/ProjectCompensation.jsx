import React from 'react'
import {browserHistory} from 'react-router'

export default class ProjectCompensation extends React.Component {
    render() {
        return (
            <div className='panel panel-info'>
                <div className="panel-heading">Panel heading</div>
                <div className="panel-body">
                    <div className="col-lg-12">
                        <span>dshjgadjksahdjkashdjkhsajkdhajkshdjkhjkdhjsak</span>
                    </div>
                    <button className="btn btn-info"
                            onClick={() =>
                                browserHistory.push(`/compensation/${this.props.compensation.id}`)}
                    >SÃ©lectionner</button>
                </div>
            </div>
        );
    }
}

