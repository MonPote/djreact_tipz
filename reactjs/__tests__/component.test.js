import React from 'react';
import renderer from 'react-test-renderer';
import ProjectCompensation from '../components/ProjectCompensation';

describe('components', function() {
  describe('<ProjectCompensation />', function() {
    it('renders correctly', function() {
      const compensation = {
          title: 'First Title',
          amount: 13,
          description: 'Description of Project Compensation',
          id: 14
      }

      var tree = renderer.create(<ProjectCompensation compensation={compensation} />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
