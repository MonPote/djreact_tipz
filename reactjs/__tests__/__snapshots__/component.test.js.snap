exports[`components <ProjectCompensation /> renders correctly 1`] = `
<div
  className="panel panel-info">
  <div
    className="panel-heading">
    First Title
     - 
    13
    € / mois
  </div>
  <div
    className="panel-body">
    <div
      className="col-lg-12">
      <span>
        Description of Project Compensation
      </span>
    </div>
    <button
      className="btn btn-info"
      onClick={[Function]}>
      Sélectionner
    </button>
  </div>
</div>
`;
